recbole/model/general_recommender/nncf.py

```diff
@@ -25,6 +25,133 @@
+                this_column_weights = np.multiply(this_column_weights, 1 / denominator)
+
+                # Sort indices and select TopK
+                # Sorting is done in three steps. Faster then plain np.argsort for higher number of items
+                # - Partition the data to extract the set of relevant items
+                # - Sort only the relevant items
+                # - Get the original item index
+                relevant_items_partition = (-this_column_weights).argpartition(self.TopK - 1)[0:self.TopK]
+                relevant_items_partition_sorting = np.argsort(-this_column_weights[relevant_items_partition])
+                top_k_idx = relevant_items_partition[relevant_items_partition_sorting]
+                i_neigh.append(top_k_idx)
+
+            start_col_block += block_size
+        # End while on columns
+
+        end_row_local = self.n_rows
+        start_row_block = 0
+
+        # Compute all similarities for each user using vectorization
+        while start_row_block < end_row_local:
+
+            end_row_block = min(start_row_block + block_size, end_row_local)
+            this_block_size = end_row_block - start_row_block
+
+            # All data points for a given user
+            user_data = self.dataMatrix[start_row_block:end_row_block, :]
+            user_data = user_data.toarray().squeeze()
+
+            if user_data.ndim == 1:
+                user_data = np.expand_dims(user_data, axis=1)
+
+            # Compute user similarities
+            this_block_weights = self.dataMatrix.dot(user_data.T)
+            for row_index_in_block in range(this_block_size):
+
+                if this_block_size == 1:
+                    this_row_weights = this_row_weights.squeeze()
+                else:
+                    this_row_weights = this_block_weights[:, row_index_in_block]
+
+                rowIndex = row_index_in_block + start_row_block
+                this_row_weights[rowIndex] = 0.0
+
+                # Apply normalization, ensure denominator != 0
+                denominator = sumOfSquared_u[rowIndex] * sumOfSquared_u + 1e-6
+                this_row_weights = np.multiply(this_row_weights, 1 / denominator)
+
+                # Sort indices and select TopK
+                # Sorting is done in three steps. Faster then plain np.argsort for higher number of users
+                # - Partition the data to extract the set of relevant users
+                # - Sort only the relevant users
+                # - Get the original user index
+                relevant_users_partition = (-this_row_weights).argpartition(self.TopK - 1)[0:self.TopK]
+                relevant_users_partition_sorting = np.argsort(-this_row_weights[relevant_users_partition])
+                top_k_idx = relevant_users_partition[relevant_users_partition_sorting]
+                u_neigh.append(top_k_idx)
+
+            start_row_block += block_size
+            # End while on rows
+
+        return u_neigh, i_neigh

```

Comment:
It's better to reuse the code from `recbole.model.general_recommender.itemknn.ComputeSimilarity`. There are too many duplicated codes.